flink-metric-prometheus/README.md

```diff
@@ -0,0 +1,81 @@
+# Overview
+
+This example shows how to report metrics defined in a FeatureView to a
+Prometheus PushGateway. It involves the following steps:
+
+1. Read a batch of historical purchase events from a file.
+
+   Each purchase event has the following fields:
+   - user_id, unique identifier of the user that made the purchase.
+   - item_id, unique identifier of the item that is purchased.
+   - price, the price of this item by the time the purchase is made.
+   - item_count, number of items purchased.
+   - timestamp, time when this purchase is made.
+
+2. For each purchase event, compute the total cost of payment made in this
+   purchase event and append it to the event as a new feature.
+   - Along with this feature, compute the number and proportion of large
+     payments (> 500) and report these metrics to a Prometheus PushGateway.
+
+3. Output the purchase events with the total cost features somewhere. As this is
+   not the focus of this example, the results are simply discarded.
+
+
+# Prerequisites
+
+Prerequisites for running this example:
+- Unix-like operating system (e.g. Linux, Mac OS X)
+- Python 3.7
+
+# Step-By-Step Instructions
+
+Please execute the following commands under the `flink-metric-prometheus` folder
+to run this example.
+
+1. Install FeatHub pip package with FlinkProcessor dependencies.
+
+   ```bash
+   $ python -m pip install --upgrade "feathub-nightly[flink]"
+   ```
+
+2. Start the Flink cluster and Prometheus PushGateway.
+
+   ```bash
+   $ docker-compose up -d
+   ```
+
+   After the Flink cluster has started, you should be able to navigate to the
+   web UI at [localhost:8081](http://localhost:8081) to view the Flink
+   dashboard.
+
+   You may also navigate to
+   [localhost:9091/metrics](http://localhost:9091/metrics) to see all the
+   metrics that has been reported to the Prometheus PushGateway. There should
+   have been some metrics reported by the pushgateway itself.
+
+3. Run the FeatHub program to compute the payment features and to report metrics
+   to the gateway.
+
+   ```bash
+   $ python main.py
+   ```
+
+4. Checkout the metrics. Apart from the metrics that have been reported since
+   the pushgateway started, you may find the following additional metrics
+   reported by the Feathub program:
+
+   ```
```

Comment:
Would it be useful to also provide the `curl` command here so that user can copy/paste/execute the command to get the expected result?